menas/src/main/scala/za/co/absa/enceladus/menas/auth/MenasAuthenticationFailureHandler.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.auth
+
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import org.slf4j.{Logger, LoggerFactory}
+import org.springframework.http.HttpStatus
+import org.springframework.security.core.AuthenticationException
+import org.springframework.security.web.authentication.AuthenticationFailureHandler
+import za.co.absa.enceladus.menas.auth.exceptions.{AuthHostTimeoutException, BadKrbHostException, BadLdapHostException}
+
+/**
+  * This class is responsible for mapping authentication exceptions to status codes in HTTP responses.
+  */
+class MenasAuthenticationFailureHandler extends AuthenticationFailureHandler {
+
+  private val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  override def onAuthenticationFailure(request: HttpServletRequest,
+                                       response: HttpServletResponse,
+                                       ex: AuthenticationException): Unit = {
+    val stack = ex.getStackTrace.mkString("\n")
+    log.warn(s"Authentication exception: ${ex.getMessage}\n$stack")
```

Comment:
```suggestion
    log.warn("Authentication exception", ex)
```
`Logger` has a smart API that allows you to log exceptions in an easy way. You don't need to construct the stack trace message yourself.